pkg/cfn/manager/nodegroup.go

```diff
@@ -28,6 +34,50 @@ func (c *StackCollection) CreateNodeGroup(seq int, errs chan error) error {
 	return c.CreateStack(name, stack, nil, errs)
 }
 
+func (c *StackCollection) ScaleNodeGroup(errs chan error) error {
+	clusterName := c.makeClusterStackName()
+	c.spec.ClusterStackName = clusterName
+	name := c.makeNodeGroupStackName(0)
+	logger.Info("scaling nodegroup stack %q in cluster %s", name, clusterName)
+
+	if c.spec.MinNodes == 0 && c.spec.MaxNodes == 0 {
+		c.spec.MinNodes = c.spec.Nodes
+		c.spec.MaxNodes = c.spec.Nodes
+	}
+
+	// Get current stack
+	template, err := c.getStackTemplate(name)
+	if err != nil {
+		return errors.Wrapf(err, "error getting stack template %s", name)
+	}
+	logger.Debug("stack template (pre-scale change): %s", template)
+
+	// Update the template
+	//TODO: In the future this needs to use Goformation but at present we manipulate the
+	// JSON directly as the version of goformation we are using isn't handling Refs well
+	var f interface{}
+	err = json.Unmarshal([]byte(template), &f)
+	if err != nil {
+		return errors.Wrap(err, "error pasring JSON")
+	}
+
+	m := f.(map[string]interface{})
```

Comment:
I once wrote https://godoc.org/github.com/errordeveloper/kubegen/pkg/macroproc#Tree, which we can extract, if seems appropriate... but there was also another package by someone else that I'm gonna try to find.

Reply:
It could had been https://github.com/tidwall/gjson, and we would need it's setter counterpart https://github.com/tidwall/sjson. Let me know what you think! I'm happy to use my own as I know it well, and it'd be worse extorting it from kubegen, but it's entirely up to you =)

Reply:
I'll take a look at them, thanks.